addons/stock/models/stock_orderpoint.py

```diff
@@ -312,16 +312,17 @@ def _get_orderpoint_action(self):
         if not to_refill:
             return action
 
-        # Remove incoming quantity from other otigin than moves (e.g RFQ)
+        # Remove incoming quantity from other origin than moves (e.g RFQ)
         product_ids, warehouse_ids = zip(*to_refill)
-        # lot_stock_ids = [lot_stock_id_by_warehouse[w] for w in warehouse_ids]
         dummy, qty_by_product_wh = self.env['product.product'].browse(product_ids)._get_quantity_in_progress(warehouse_ids=warehouse_ids)
         rounding = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+        # Group orderpoint by product-warehouse
+        orderpoint_by_product_warehouse = defaultdict(OrderedSet)
+        for orderpoint in orderpoints:
+            orderpoint_by_product_warehouse[(orderpoint.product_id.id, orderpoint.warehouse_id.id)].add(orderpoint.id)
```

Comment:
Maybe better use `read_group`
<pre>
grouped_orderpoint_data = self.env['stock.warehouse.orderpoint'].read_group(
            [('id', 'in', orderpoints.ids)],
            ['product_id', 'warehouse_id', 'qty_to_order'],
            ['product_id', 'warehouse_id'], lazy=False)
orderpoint_by_product_warehouse = {
    (record.get('product')[0], record.get('warehouse')[0]): record.get('qty_to_order')
    for record in grouped_orderpoint_data
}
</pre>

Reply:
Yes, good point. thx.